internal/rbd/rbd_attach.go

```diff
@@ -462,9 +464,24 @@ func detachRBDImageOrDeviceSpec(
 	}
 	if dArgs.isNbd && dArgs.logDir != "" {
 		logFile := getCephClientLogFileName(dArgs.volumeID, dArgs.logDir, "rbd-nbd")
-		if err = os.Remove(logFile); err != nil {
-			log.WarningLog(ctx, "failed to remove logfile: %s, error: %v",
-				logFile, err)
+		switch logStrategyLookup[dArgs.logStatergy] {
+		case compress:
+			if err = log.GzipLogFile(logFile); err != nil {
```

Comment:
the time taken here depends on the file size? if yes can we do it in go routine to avoid affecting the GRPC call performance?